src/testing/tree_construction.rs

```diff
@@ -445,7 +447,11 @@ fn document(s: &str) -> Vec<String> {
                 Some(format!("|{}", l.trim_end()))
             }
         })
-        .collect::<Vec<_>>()
+        .collect::<Vec<_>>();
+
+    // TODO: drop the following line
+    document.push("".into());
+    document
```

Comment:
It looks like some of the newer tests to be parsed are relying (perhaps incorrectly) on there being an extra blank string at the end of the document array.